sql/src/main/java/io/crate/analyze/CopyToAnalyzedStatement.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.crate.analyze.relations.AnalyzedRelation;
+import io.crate.analyze.relations.AnalyzedRelationVisitor;
+import io.crate.analyze.relations.QueriedDocTable;
+import io.crate.analyze.symbol.Field;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.exceptions.ColumnUnknownException;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.Path;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+public class CopyToAnalyzedStatement extends AbstractCopyAnalyzedStatement implements AnalyzedRelation {
+
+    private QueriedDocTable subQueryRelation;
+    private boolean columnsDefined;
+    private Map<ColumnIdent, Symbol> overwrites = ImmutableMap.of();
+
+    public void subQueryRelation(QueriedDocTable subQueryRelation) {
```

Comment:
same here.. I think it would be possible to change this class to have a constructor which takes all arguments instead of setters